akka-stream/src/main/scala/akka/stream/impl/TraversalBuilder.scala

```diff
@@ -267,11 +267,8 @@ sealed trait TraversalBuilder {
   protected def internalSetAttributes(attributes: Attributes): TraversalBuilder
 
   def setAttributes(attributes: Attributes): TraversalBuilder = {
-    if (attributes ne Attributes.none) {
-      if (attributes.isAsync) this.makeIsland(GraphStageTag).internalSetAttributes(attributes)
-      else internalSetAttributes(attributes)
-    } else
-      this
+    if (attributes.isAsync) this.makeIsland(GraphStageTag).internalSetAttributes(attributes)
```

Comment:
um, sounds like a weird thing to do, unless it's needed for fusing together a stage with IODispatcher as default together with a simple map stage. Shall we create a ticket about how to define async islands (might need some investigation of how it works and clarifications in documentation)?